nss_cache/sources/ldapsource.py

```diff
@@ -397,9 +397,15 @@ def __iter__(self):
         # If the dn is requested, return it along with the payload,
         # otherwise ignore it.
         for key in record[1]:
-          if isinstance(record[1][key][0], bytes) and key != 'objectSid':
-            value = record[1][key][0].decode('utf-8')
-            record[1][key] = [value]
+          if len(record[1][key]) > 1:
```

Comment:
Avoid the `if` statement here, by using a temporary list we can append to, then finally assign record[1][key] = tmp_list.

Reply:
this is possible as well.